File: 783/minimum-distance-between-bst-nodes.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function (root) {

    let vals = [];
    function dfs(node) {
        if (!node) return;
        if (node.left)
            dfs(node.left);
        vals.push(node.val);
        if (node.right)
            dfs(node.right);
    }

    dfs(root);
    let minDiff = Infinity;
    for (let i = 1; i < vals.length; i++) {
        minDiff = Math.min(minDiff, vals[i] - vals[i - 1]);
    }
    return minDiff;
};

File: 783/minimum-distance-between-bst-nodes.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func dfs(node *TreeNode, values *[]int) {
	if node == nil {
		return
	}

	dfs(node.Left, values)
	*values = append(*values, node.Val)
	dfs(node.Right, values)
}
func minDiffInBST(root *TreeNode) int {
	vals := []int{}
	dfs(root, &vals)
	minDiff := vals[1] - vals[0]
	for i := 1; i < len(vals); i++ {
		if vals[i]-vals[i-1] < minDiff {
			minDiff = vals[i] - vals[i-1]
		}
	}
	return minDiff
}